sky/spot/controller.py

```diff
@@ -24,6 +25,15 @@
 
 logger = sky_logging.init_logger(__name__)
 
+# We start process with 'spawn', because 'fork' results in weird
+# behaviors like ignoring SIGINT by the controller process when
+# it is started inside a Ray job. The previous fix was using and
+# catching SIGTERM instead, but recently I find this would crash the
+# process after ~10s, so this is clearly not a correct solution.
+# The correct solution uses 'spawn' to start a new process and
+# just send SIGINT for interruption.
```

Comment:
nit:
```suggestion
# We start process with 'spawn', because 'fork' results in weird
# behaviors like ignoring SIGINT by the controller process when
# it is started inside a Ray job. The previous fix sent the SIGTERM
# and override the handle to raise KeyboardInterrupt, but that would
# crash the process after ~10s, i.e., if the termination of the cluster
# does not finish in 10s, the spot cluster will be leaked. 'spawn' solves
# the problem by making the SIGINT work.
```